### PR TITLE
New 404 page

### PR DESCRIPTION
**Pull request by arshdkhn1**
image

### CODE REVIEW

File: lib/routes.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ const landingSection = camp.template([
   './templates/projects.html',
 ]);
 exports.landingPage = function (response, user = null) {
-  const title = '';
+  const title = 'Janitor';
```
Comment (arshdkhn1) [inline lib/routes.js]: See header to know why i did this.

File: static/css/janitor-new.css [hunk 2]
```diff
@@ -296,6 +302,11 @@ p {
   margin: 0;
 }
 
+/* 404 page */
+section.page-not-found-section {
+  margin: auto;
```
Comment (nt1m) [inline static/css/janitor-new.css]: You seem to be using this in your login page and here too. Can you introduce a new class for centering both sides  and remove the `.page-not-found-section` class ? `.centered-container`

File: templates/404-new.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+    <section class="text-center page-not-found-section">
```
Comment (nt1m) [inline templates/404-new.html]: Please use a `div`. `section` is going to be used by the settings page for different setting sections

File: lib/routes.js [hunk 4]
```diff
@@ -455,6 +454,24 @@ exports.notFoundPage = function (response, user) {
...
 
+// New 404 Not Found page.
+const newNotFoundSection = camp.template('./templates/404-new.html');
+exports.newNotFoundPage = function (response, user) {
+  const title = "Page not found"
+  const pageNotFound = true;
+  response.statusCode = 404;
+  response.template({
+    title,
+    pageNotFound,
```
Comment (arshdkhn1) [inline lib/routes.js]: I made an extra variable to avoid string comparison in header.

File: templates/header-new.html [hunk 5]
```diff
@@ -5,12 +5,13 @@
     <meta name="application-name" content="Janitor">
     <meta name="theme-color" content="#ffffff">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <title>{{= title in html}}{{if title then {{ – }}}}Janitor</title>
+    <title>{{= title in html}}{{if {{title && !pageNotFound}} then {{ – Janitor}}}}</title>
```
Comment (arshdkhn1) [inline templates/header-new.html]: To make a clean 404 page.

File: templates/header-new.html [hunk 6]
```diff
@@ -5,12 +5,13 @@
...
     <link href="/manifest.json" rel="manifest">
     <link href="/css/janitor-new.css" rel="stylesheet">
     <link href="/img/janitor.svg" rel="icon" sizes="any" type="image/svg+xml">
   </head>
   <body>
+    {{if !pageNotFound then {{
```
Comment (arshdkhn1) [inline templates/header-new.html]: To make a clean 404 page.

File: templates/footer-new.html [hunk 7]
```diff
@@ -30,10 +30,11 @@
         </div>
       </div>
     </footer>
+    {{if !pageNotFound then {{
```
Comment (nt1m) [inline templates/footer-new.html]: Let's just load all the scripts for now, other pages like the blog page are affected by this too. So I'd prefer coming up with a proper solution for all pages.

File: lib/routes.js [hunk 8]
```diff
@@ -455,6 +455,25 @@ exports.notFoundPage = function (response, user) {
   ]);
 };
 
+// New 404 Not Found page.
+const newNotFoundSection = camp.template('./templates/404-new.html');
+exports.newNotFoundPage = function (response, user) {
```
Comment (nt1m) [inline lib/routes.js]: nit: the convention is `notFoundSectionNew` and `notFoundPageNew`

File: lib/routes.js [hunk 9]
```diff
@@ -455,6 +455,25 @@ exports.notFoundPage = function (response, user) {
...
+  const title = "Page not found"
+  const pageNotFound = true;
+
+  response.statusCode = 404;
+  response.template({
+    title,
+    pageNotFound,
```
Comment (nt1m) [inline lib/routes.js]: nit: remove this pageNotFound var.

File: lib/routes.js [hunk 10]
```diff
@@ -455,6 +455,25 @@ exports.notFoundPage = function (response, user) {
   ]);
 };
 
+// New 404 Not Found page.
+const newNotFoundSection = camp.template('./templates/404-new.html');
+exports.newNotFoundPage = function (response, user) {
+  const title = "Page not found"
```
Comment (nt1m) [inline lib/routes.js]: nit: semi colon (eslint will detect this)

File: static/css/janitor-new.css (hunk 2 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: yes, that i was thinking but I thought i would let it be a part of next pr. I thought you might not like main tag having flex display.

File: templates/404-new.html (hunk 3 above)
Comment (arshdkhn1) [reply templates/404-new.html]: cool

File: templates/footer-new.html (hunk 7 above)
Comment (arshdkhn1) [reply templates/footer-new.html]: how is blog page affected by this?

File: lib/routes.js (hunk 8 above)
Comment (arshdkhn1) [reply lib/routes.js]: ohk

File: lib/routes.js (hunk 10 above)
Comment (arshdkhn1) [reply lib/routes.js]: good catch

File: templates/footer-new.html [hunk 11]
```diff
@@ -29,7 +29,7 @@
           <p class="footer-logo-tagline">The fastest development system in the world.</p>
         </div>
       </div>
-    </footer>
+    </footer>
```
Comment (nt1m) [inline templates/footer-new.html]: nit: whitespace

File: lib/routes.js [hunk 12]
```diff
@@ -438,7 +438,7 @@ exports.adminPage = function (response, section, user) {
   ]);
 };
 
-// 404 Not Found page.
+// notFoundSectionNew Not Found page.
```
Comment (nt1m) [inline lib/routes.js]: nit: undo this change

File: app.js [hunk 13]
```diff
@@ -447,6 +447,13 @@ boot.executeInParallel([
       });
     });
   });
+  
+  // New 404 Not Found page
+  app.route(/^\/404-new$/, (data, match, end, query) => {
```
Comment (jankeromnes) [inline app.js]: Nit: Please add `\/?` before the `$` to support a trailing slash.

File: app.js [hunk 14]
```diff
@@ -447,6 +447,13 @@ boot.executeInParallel([
...
+    const { user } = query.req;
+    log('404', match[0]);
```
Comment (jankeromnes) [inline app.js]: Nit: '404-new'?

File: lib/routes.js [hunk 15]
```diff
@@ -455,6 +455,23 @@ exports.notFoundPage = function (response, user) {
   ]);
 };
 
+// New 404 Not Found page.
+const notFoundSectionNew = camp.template('./templates/404-new.html');
+exports.notFoundPageNew = function (response, user) {
+  const title = "Page not found";
```
Comment (jankeromnes) [inline lib/routes.js]: Nit: Please use single quotes to stay consistent with the rest of the code base.

File: static/css/janitor-new.css [hunk 16]
```diff
@@ -81,19 +81,29 @@ h4 {
 main {
   overflow: auto;
   flex-grow: 1;
-  /* viewport height - 15px extra height(so that footer is visible) */
-  min-height: calc(100vh - 15px);
+  /* viewport height - header height - 15px extra height(so that footer is visible) */
```
Comment (jankeromnes) [inline static/css/janitor-new.css]: Nit: You can add a space before the `(`.

**Comment by jankeromnes** [comment]
Ah, oops, it was merged already. You win! :)

@arshdkhn1 please have a look at my nits, and address them in another PR if you think they make sense.

File: app.js (hunk 13 above)
Comment (nt1m) [reply app.js]: I personally don't mind this too much as it's going to be merged with the default 404 page at some point...
